src/ui/QuerySummary/QuerySummary.ts

```diff
@@ -189,19 +234,123 @@ export class QuerySummary extends Component {
     }
   }
 
+  private updateNoResultsContainerSnapshot() {
```

Comment:
The question is about when should we update the snapshot. 

Technically, if we just consider the scenario when we are in the IE we could just verify if the container exists (`if (noResultsContainer)`) because each time we make some changes to the QuerySummary node via the editor the node's constructor gets called and resets the `noResultsContainerSnapshot` to undefined.

But, in the scenario where some custom Javascript updates the content of the node I'm not sure if the constructor is gonna be recalled in every scenario. (Probably not, so how should we handle the snapshot's update?)

_Note : `this.noResultsContainerSnapshot == null` is probably not needed and/or the good logic to have._